runtime/near-vm-logic/src/tests/gas_counter.rs

```diff
@@ -96,12 +96,76 @@ fn test_hit_prepaid_gas_limit() {
 
     promise_create(&mut logic, b"rick.test", 0, gas_limit / 2).expect("should create a promise");
     logic.gas(op_limit * 2).expect_err("should fail with gas limit");
-    let outcome = logic.outcome();
+    let outcome = logic.compute_outcome_and_distribute_gas();
 
     assert_eq!(outcome.burnt_gas, gas_limit);
     assert_eq!(outcome.used_gas, gas_limit);
 }
 
+#[cfg(feature = "protocol_feature_function_call_weight")]
+fn function_call_weight_check(function_calls: impl IntoIterator<Item = (u64, u64)>) {
+    use near_primitives::types::GasWeight;
+
+    let gas_limit = 10u64.pow(14);
+
+    let mut logic_builder = VMLogicBuilder::default().max_gas_burnt(gas_limit);
+    let mut logic = logic_builder.build_with_prepaid_gas(gas_limit);
+
+    let mut weight_sum = 0u128;
+    for (static_gas, gas_weight) in function_calls {
+        let index = promise_batch_create(&mut logic, "rick.test").expect("should create a promise");
+        promise_batch_action_function_call_weight(
+            &mut logic,
+            index,
+            0,
+            static_gas,
+            GasWeight(gas_weight),
+        )
+        .expect("batch action function call should succeed");
+
+        weight_sum += gas_weight as u128;
+    }
+    let outcome = logic.compute_outcome_and_distribute_gas();
+
+    // Verify that all gas is used when only one ratio is specified
+    println!("{} {} {}", outcome.used_gas, gas_limit, weight_sum);
+    assert!(outcome.used_gas as u128 + weight_sum - 1 >= gas_limit as u128);
+}
```

Comment:
should we also check that used_gas < gas_limit? 

And maybe return how much each promise got? (so that we can verify it below)